static/market/js/utils.js

```diff
@@ -1,167 +1,186 @@
-    }
-
-    // Return time from now data
-    return `${tfn.time} ${tfn.unitOfTime}`
+  // Get timestamps
+  let unixTime = new Date(time).getTime();
+  if (!unixTime) return;
+  let now = new Date().getTime();
+
+  // Calculate difference
+  let difference = unixTime / 1000 - now / 1000;
+
+  // Setup return object
+  let tfn = {};
+
+  // Check if time is in the past, present, or future
+  tfn.when = "now";
+  if (difference > 0) {
+    tfn.when = "future";
+  } else if (difference < -1) {
+    tfn.when = "past";
+  }
+
+  // Convert difference to absolute
+  difference = Math.abs(difference);
+
+  // Calculate time unit
+  if (difference / (60 * 60 * 24 * 365) > 1) {
+    // Years
+    tfn.unitOfTime = "years";
+    tfn.time = Math.floor(difference / (60 * 60 * 24 * 365));
+  } else if (difference / (60 * 60 * 24 * 45) > 1) {
+    // Months
+    tfn.unitOfTime = "months";
+    tfn.time = Math.floor(difference / (60 * 60 * 24 * 45));
+  } else if (difference / (60 * 60 * 24) > 1) {
+    // Days
+    tfn.unitOfTime = "days";
+    tfn.time = Math.floor(difference / (60 * 60 * 24));
+  } else if (difference / (60 * 60) > 1) {
+    // Hours
+    tfn.unitOfTime = "hours";
+    tfn.time = Math.floor(difference / (60 * 60));
+  } else if (difference / 60 > 1) {
+    // Minutes
+    tfn.unitOfTime = "minutes";
+    tfn.time = Math.floor(difference / 60);
+  } else {
+    // Seconds
+    tfn.unitOfTime = "seconds";
+    tfn.time = Math.floor(difference);
+  }
+
+  // Return time from now data
+  return `${tfn.time} ${tfn.unitOfTime}`;
 }
 
 function isValidImageUrl(string) {
-    let url
-    try {
-        url = new URL(string)
-    } catch (_) {
-        return false
-    }
-    return url.protocol === 'http:' || url.protocol === 'https:'
+  let url;
+  try {
+    url = new URL(string);
+  } catch (_) {
+    return false;
+  }
+  return url.protocol === "http:" || url.protocol === "https:";
 }
 
-function isValidKey(key, prefix = 'n') {
-    try {
-        if (key && key.startsWith(prefix)) {
-            let { _, data } = NostrTools.nip19.decode(key)
-            key = data
-        }
-        return isValidKeyHex(key)
-    } catch (error) {
-        return false
+function isValidKey(key, prefix = "n") {
+  try {
+    if (key && key.startsWith(prefix)) {
+      let { _, data } = NostrTools.nip19.decode(key);
+      key = data;
     }
+    return isValidKeyHex(key);
+  } catch (error) {
+    return false;
+  }
 }
 
 function isValidKeyHex(key) {
-    return !!key?.toLowerCase()?.match(/^[0-9a-f]{64}$/)
+  return !!key?.toLowerCase()?.match(/^[0-9a-f]{64}$/);
 }
 
 function formatCurrency(value, currency) {
+  try {
     return new Intl.NumberFormat(window.LOCALE, {
-        style: 'currency',
-        currency: currency
-    }).format(value)
+      style: "currency",
+      currency: currency,
+    }).format(value);
+  } catch (error) {
+    return value;
+  }
+}
+
+function productCompare(a, b, sortBy, sortOrder) {
```

Comment:
why was this added?

Reply:
I didn't modify anything.

looks like it was done in lnbits/nostr-market-app#11